lib/internal/streams/transform.js

```diff
@@ -82,6 +84,26 @@ function Transform(options) {
   if (!(this instanceof Transform))
     return new Transform(options);
 
+  // TODO (ronag): This should preferably always be
+  // applied but would be semver-major. Or even better
+  // Make transform a Readable with the Writable interface.
+  if (options && options.highWaterMark === 0) {
+    // A Duplex will buffer both on the writable and readable side while
+    // a Transform just wants to buffer hwm number of elements. To avoid
+    // buffering twice we disable buffering on the writable side.
```

Comment:
Would it not make more sense to buffer half on the writable end?

Reply:
Per my comments in PR. A transform should just be a normal readable that implements the writable interface. Since we are not there yet I think a suitable step towards that direction is to disable the writable as much as possible.